### PR TITLE
Display version status in `Audit Vulnerabilities` and `Exploit Predictions` tab

### PR DESCRIPTION
**Pull request by rbt-mm**
### Description

Information about the version status of a component is currently only displayed in the `Components` tab of a project.

Also adding this information to the `Audit Vulnerabilities` and the `Exploit Predictions` tab of a project improves the usability of those two tabs, because it is no longer needed to switch tabs to check for version status information.

### Addressed Issue

Frontend: [#356](https://github.com/DependencyTrack/frontend/issues/356)

### Additional Details

The retrieved `Finding` objects from `/finding/project/<projectUUID>` must be appended by the latest version of a component to correctly display the information in the UI.

[Frontend PR](https://github.com/DependencyTrack/frontend/pull/357)

image

image

### Checklist

- [x] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/github-templates/CONTRIBUTING.md#pull-requests)
~- [ ] This PR fixes a defect, and I have provided tests to verify that the fix is effective~
~- [ ] This PR implements an enhancement, and I have provided tests to verify that it works as intended~
~- [ ] This PR introduces changes to the database model, and I have added corresponding [update logic](https://github.com/DependencyTrack/dependency-track/tree/master/src/main/java/org/dependencytrack/upgrade)~
~- [ ] This PR introduces new or alters existing behavior, and I have updated the [documentation](https://github.com/DependencyTrack/dependency-track/tree/master/docs/_docs) accordingly~

### CODE REVIEW

File: src/main/java/org/dependencytrack/persistence/FindingsQueryManager.java [hunk 1]
```diff
@@ -278,6 +281,14 @@ public List<Finding> getFindings(Project project, boolean includeSuppressed) {
...
                 finding.getVulnerability().put("description", vulnerability.getDescription());
                 finding.getVulnerability().put("recommendation", vulnerability.getRecommendation());
+                final PackageURL purl = component.getPurl();
+                if (purl != null) {
+                    final RepositoryType type = RepositoryType.resolve(purl);
+                    if (RepositoryType.UNSUPPORTED != type) {
+                        final RepositoryMetaComponent repoMetaComponent = getRepositoryMetaComponent(type, purl.getNamespace(), purl.getName());
+                        finding.getComponent().put("latestVersion", repoMetaComponent.getLatestVersion());
+                    }
+                }
```
Comment (nscuro) [inline src/main/java/org/dependencytrack/persistence/FindingsQueryManager.java:291]: It should be possible to to this via `LEFT JOIN` in the existing findings query:

https://github.com/DependencyTrack/dependency-track/blob/8f4c2b5c0d2e153363fed0ab2e865653558b7d6d/src/main/java/org/dependencytrack/model/Finding.java#L50-L81

That would save us a few database round-trips.

**Review by nscuro — CHANGES_REQUESTED** [review]

File: src/main/java/org/dependencytrack/persistence/FindingsQueryManager.java (hunk 1 above)
Comment (rbt-mm) [reply src/main/java/org/dependencytrack/persistence/FindingsQueryManager.java:291]: The problem that I see here is that neither `COMPONENT` nor `REPOSITORY_META_COMPONENT` reference each other via a foreign key.

**Comment by nscuro** [comment]
@rbt-mm Could you add a test for this in `FindingResourceTest`? Also, please update the example in the FPF documentation to include the new field: https://github.com/DependencyTrack/dependency-track/blob/master/docs/_docs/integrations/file-formats.md

**Review by nscuro — APPROVED** [review]
Thanks @rbt-mm! 🚀

**Pull request merged**